src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java

```diff
@@ -120,24 +120,24 @@ public String getApiUri() {
     @Override public void visitSources(SCMSourceObserver observer) throws IOException, InterruptedException {
         TaskListener listener = observer.getListener();
         if (repoOwner.isEmpty()) {
-            listener.getLogger().format("Must specify user or organization%n");
-            return;
+            throw new AbortException("Must specify user or organization%n");
         }
         StandardCredentials credentials = Connector.lookupScanCredentials(observer.getContext(), apiUri, scanCredentialsId);
         GitHub github = Connector.connect(apiUri, credentials);
         if (credentials != null && !github.isCredentialValid()) {
-            listener.getLogger().format("Invalid scan credentials %s to connect to %s, skipping%n", CredentialsNameProvider.name(credentials), apiUri == null ? "github.com" : apiUri);
-            return;
+            String message = String.format("Invalid scan credentials %s to connect to %s, skipping%n", CredentialsNameProvider.name(credentials), apiUri == null ? "github.com" : apiUri);
+            throw new AbortException(message);
         }
 
         if (!github.isAnonymous()) {
             listener.getLogger().format("Connecting to %s using %s%n", apiUri == null ? "github.com" : apiUri, CredentialsNameProvider.name(credentials));
-            GHMyself myself = null;
+            GHMyself myself = github.getMyself();
             try {
                 // Requires an authenticated access
                 myself = github.getMyself();
-            } catch (IOException e) {
-                // Something wrong happened, maybe java.net.ConnectException?
+            } catch (RateLimitExceededException rle) {
+                listener.getLogger().format("%n%s%n%n", rle.getMessage());
+                throw new InterruptedException();
```

Comment:
Sure, this was copied from [here](https://github.com/recena/github-branch-source-plugin/blob/83fd196b646cae4e7d99d6b4e0c7bf5e2680a014/src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMNavigator.java#L159-159). In fact, we decided to use `AbortException` instead of `InterruptedException`.